01_prepare_host.sh

```diff
@@ -38,20 +38,36 @@ if ! which kustomize 2>/dev/null ; then
     sudo mv kustomize /usr/local/bin/.
 fi
 
-# Download Ironic binary and Ironic inspector image
+# Clean-up any old ironic containers
+for name in ironic ironic-inspector dnsmasq httpd mariadb ipa-downloader; do
+    sudo "${CONTAINER_RUNTIME}" ps | grep -w "$name$" && sudo "${CONTAINER_RUNTIME}" kill $name
+    sudo "${CONTAINER_RUNTIME}" ps --all | grep -w "$name$" && sudo "${CONTAINER_RUNTIME}" rm $name -f
+done
 
-mkdir -p "$IRONIC_DATA_DIR/html/images"
-pushd "$IRONIC_DATA_DIR/html/images"
-if [ ! -f ironic-python-agent.initramfs ]; then
-    curl --insecure --compressed -L https://images.rdoproject.org/master/rdo_trunk/current-tripleo-rdo/ironic-python-agent.tar | tar -xf -
+if [[ "${CONTAINER_RUNTIME}" == "podman" ]]; then
+  # Remove existing pod
+  if  sudo "${CONTAINER_RUNTIME}" pod exists ironic-pod ; then
+      sudo "${CONTAINER_RUNTIME}" pod rm ironic-pod -f
+  fi
+  # Create pod
```

Comment:
let's pause here? At least a 1 second sleep would be nice, just to give the time to the system to process the precedent change

Reply:
This is `docker ps` (or `podman ps`) not ps itself

Reply:
Why sleep? dev-scripts doesn't sleep for the same code, and I don't see any evidence `podman pod rm` is asynchronous

Reply:
Yeah I agree, random sleeps in scripts are poor practice IMHO, if there's something we actually need to wait for then we should do a loop, podman wait or whatever, but as mentioned we don't wait in dev-scripts so this is probably fine as-is.

`man podman-pod-rm` doesn't actually document the exit status, but podman-rm says exit code zero if all containers are removed, so I'd assume podman pod rm works the same, e.g we should fail if the pod isn't removed, assuming set -e

Reply:
I'm totally up for a more robust approach
again, can be done in a follow up